raiden/network/proxies/token_network.py

```diff
@@ -162,20 +176,94 @@ def new_netting_channel(
         if self.node_address == partner:
             raise SamePeerAddress('The other peer must not have the same address as the client.')
 
+        channel_exists = self.channel_exists_and_not_settled(
+            participant1=self.node_address,
+            participant2=partner,
+            block_identifier=block_identifier,
+        )
+        if channel_exists:
+            raise DuplicatedChannelError('Channel with given partner address already exists')
+
+    def _new_channel_postconditions(
+            self,
+            partner: Address,
+            block: BlockSpecification,
+    ) -> Tuple[bool, str]:
+        channel_created = self.channel_exists_and_not_settled(
+            participant1=self.node_address,
+            participant2=partner,
+            block_identifier=block,
+        )
+        if channel_created:
+            return True, 'Channel with given partner address already exists'
+        return False, ''
+
+    def new_netting_channel(
+            self,
+            partner: Address,
+            settle_timeout: int,
+    ) -> ChannelID:
+        """ Creates a new channel in the TokenNetwork contract.
+
+        Args:
+            partner: The peer to open the channel with.
+            settle_timeout: The settle timeout to use for this channel.
+
+        Returns:
+            The ChannelID of the new netting channel.
+        """
+
+        self._new_channel_preconditions(partner, settle_timeout, 'pending')
         log_details = {
             'peer1': pex(self.node_address),
             'peer2': pex(partner),
         }
-        log.debug('new_netting_channel called', **log_details)
+        gas_limit = self.proxy.estimate_gas(
```

Comment:
this should be done once the lock is held, otherwise the gas estimation *may be* wrong once the transaction is sent.

Reply:
Which lock? The `self.channel_operations_lock[partner]` lock? So far (also in master) we don't acquire that lock during channel creation.

Reply:
hummm, true, I didn't realize that open was not acquiring the lock. Is this correct?